v2/checkSemver.js

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env node
+
+const fs = require("fs/promises");
+const path = require("path");
+const YAML = require("yaml");
+const axios = require("axios");
+const semver = require("semver");
+
+Promise.all([
+  axios
+    .get(
+      "https://api.github.com/repos/ubports/ubports-installer/releases/latest"
+    )
+    .then(({ data }) => data.tag_name),
+  fs
+    .readdir("v2/devices")
+    .then(files =>
+      Promise.all(
+        files.map(f =>
+          fs
+            .readFile(path.join("v2", "devices", f))
+            .then(cfg => YAML.parse(cfg.toString()))
+            .then(({ codename, operating_systems }) =>
+              operating_systems
+                .map(({ compatible_installer, name }) => ({
+                  codename,
+                  name,
+                  compatible_installer
+                }))
+                .filter(({ compatible_installer }) => compatible_installer)
+            )
+        )
+      )
+    )
+    .then(devices => devices.flat())
+])
+  .then(([latest_installer, devices]) =>
+    Promise.all(
+      devices.map(device =>
+        semver.satisfies(latest_installer, device.compatible_installer, {
+          includePrerelease: true
+        })
+          ? Promise.resolve()
+          : Promise.reject(
+              new Error(
+                `${device.codename}: ${device.name}: compatible_installer ${device.compatible_installer} does not satisfy latest installer version ${latest_installer}`
```

Comment:
Do I understand right: This means that before an installer config can be merged at least one compatible installer version must exist in the releases section of installer repo.

Reply:
no. it gets the latest release and checks if everything is compatible.

Reply:
Yes. What I mean is: If I create now an installer config that requires 0.9.4 it couldnt be merged until this installer is released.

Reply:
yeah. right, i should probably not do it like this :]

Reply:
Well it depends: If we want a 100% working state its ok to require it. My thought was just to question it, not to turn it down. If we go that way, we should just know this and also tell the world that this behaviour changes.

Reply:
I'm all for leaving this in, one less opportunity for a typo or miscommunication

Reply:
i'm leaving the script in, but won't assert it's passing for ci to greenlight a PR. that way you can still check manually if you want